File: samples/Julia/anagrams.jl
url = "http://www.puzzlers.org/pub/wordlists/unixdict.txt"

wordlist = map!(chomp,(open(readlines, download(url)))) ;

function anagram(wordlist)
  hash = Dict() ; ananum = 0
  for word in wordlist
    sorted = CharString(sort(collect(word.data)))
    hash[sorted] = [ get(hash, sorted, {}), word ]
    ananum = max(length(hash[sorted]), ananum)
  end
  collect(values(filter((x,y)-> length(y) == ananum, hash)))
end

File: samples/PHP/anagrams.php
<?php
$words = explode("\n", file_get_contents('http://www.puzzlers.org/pub/wordlists/unixdict.txt'));
foreach ($words as $word) {
    $chars = str_split($word);
    sort($chars);
    $anagram[implode($chars)][] = $word;
}

$best = max(array_map('count', $anagram));
foreach ($anagram as $ana)
    if (count($ana) == $best)
        print_r($ana);
?>
